### PR TITLE
Data normalization apis [Genus api + Species api]

### PR DESCRIPTION
**Pull request by RajKumar-BackCountry**

### CODE REVIEW

File: src/modules/dataNormalization/dataNormalization.js [hunk 1]
```diff
@@ -52,39 +52,144 @@ router.get('/styles/:styleId/techSpecs', async (req, res) => {
...
+      }
+    });
   } catch (error) {
-    console.error(error.message);
-    res.sendResponse('Internal Server Error', 500);
+    return res.sendResponse('Internal Server Error', 500);
   }
 });
 
 router.get('/genus/:genusId/species', async (req, res) => {
```
Comment (nikhilchandrabackcountry) [inline src/modules/dataNormalization/dataNormalization.js:138]: join the database tables to get the data

File: src/modules/dataNormalization/dataNormalization.js [hunk 2]
```diff
@@ -52,39 +52,144 @@ router.get('/styles/:styleId/techSpecs', async (req, res) => {
...
+        id: parseInt(genusId)
+      },
+      select: {
+        id: true,
+        dattributelid: true
+      }
+    });
+
+    if (genus.dattributelid) {
+      const dn_dattributev = await postgresPrisma.dn_dattributev.findMany({
```
Comment (nikhilchandrabackcountry) [inline src/modules/dataNormalization/dataNormalization.js]: camelcase

**Review by nikhilchandrabackcountry — CHANGES_REQUESTED** [review]

**Comment by nikhilchandrabackcountry** [comment]
do npx eslint . --fix, there are some errors

File: src/modules/dataNormalization/dataNormalization.js [hunk 3]
```diff
@@ -52,39 +52,160 @@ router.get('/styles/:styleId/techSpecs', async (req, res) => {
...
+        total,
+        pageCount,
+        currentPage: parsedPage,
+        currentPageCount: genus.length
+      }
+    });
   } catch (error) {
-    console.error(error.message);
-    res.sendResponse('Internal Server Error', 500);
+    return res.sendResponse('Internal Server Error', 500);
```
Comment (nikhilchandrabackcountry) [inline src/modules/dataNormalization/dataNormalization.js]: error while fecthing *******

File: src/modules/dataNormalization/dataNormalization.js [hunk 4]
```diff
@@ -52,39 +52,160 @@ router.get('/styles/:styleId/techSpecs', async (req, res) => {
...
+//       }
+//     });
+//     return res.sendResponse(hAttributes);
+//   } catch (error) {
+//     console.error(error.message);
+//     res.send('Internal Server Error').status(500);
+//   }
+// });
+
 router.get('/genus', async (req, res) => {
```
Comment (nikhilchandrabackcountry) [inline src/modules/dataNormalization/dataNormalization.js:86]:
```suggestion
router.get('/genus', async (req, res) => {
  try {
    const { page, limit = 10 } = req.query;
    const parsedLimit = parseInt(limit, 10);

    if (!page) {
      const genus = await postgresPrisma.dn_genus.findMany({});
      const total = genus.length;
      const pageCount = Math.ceil(total / parsedLimit);

      return res.sendResponse({
        genus,
        pagination: {
          total,
          pageCount,
          currentPage: 1,
          currentPageCount: genus.length
        }
      });
    }

    const parsedPage = parseInt(page, 10);

    if (Number.isNaN(parsedLimit) || Number.isNaN(parsedPage)) {
      return res.sendResponse('Invalid page or limit value. Please provide valid numeric values for page and limit parameters.', 400);
    }

    const skip = (parsedPage - 1) * parsedLimit;

    const [genus, total] = await Promise.all([
      postgresPrisma.dn_genus.findMany({
        skip,
        take: parsedLimit
      }),
      postgresPrisma.dn_genus.count()
    ]);

    const pageCount = Math.ceil(total / parsedLimit);

    return res.sendResponse({
      genus,
      pagination: {
        total,
        pageCount,
        currentPage: parsedPage,
        currentPageCount: genus.length
      }
    });
  } catch (error) {
    console.error(error);
    return res.sendResponse('An error occurred while retrieving the genus data. Please try again later.', 500);
  }
});

```

File: src/modules/dataNormalization/dataNormalization.js (hunk 1 above)
Comment (RajKumar-BackCountry) [reply src/modules/dataNormalization/dataNormalization.js:138]: Can u clarify ?

File: src/modules/dataNormalization/dataNormalization.js (hunk 4 above)
Comment (nikhilchandrabackcountry) [reply src/modules/dataNormalization/dataNormalization.js:86]: check this code out

File: src/modules/dataNormalization/dataNormalization.js [hunk 5]
```diff
@@ -52,39 +52,160 @@ router.get('/styles/:styleId/techSpecs', async (req, res) => {
...
+        id: parseInt(genusId)
+      },
+      select: {
+        id: true,
+        dattributelid: true
+      }
+    });
+
+    if (genus.dattributelid) {
+      const dAttributes = await postgresPrisma.dn_dattributev.findMany({
```
Comment (nikhilchandrabackcountry) [inline src/modules/dataNormalization/dataNormalization.js]: join dn_genus and dn_dattributev and get the desired data

Comment (nikhilchandrabackcountry) [reply src/modules/dataNormalization/dataNormalization.js]: also this dn_genus_attributel_attributetype

**Review by nikhilchandrabackcountry — APPROVED** [review]

**Pull request merged**